### PR TITLE
test: Add CodeCov reporting for the Profiler

### PR DESCRIPTION
**Pull request by chynesNR**
* The "Build and Package Profiler" GitHub Action now also runs the Profiler's unit tests and uploads the results to CodeCov
* Added a "Deploy" checkbox to that job (checked by default). You can uncheck this to generate a new CodeCov report without publishing a new Profiler build
* Added an "Agent" flag to our Agent's CodeCov report, so that we can view it separately from the Profiler's coverage
* Increased code coverage for the Profiler from 65% to 75%

### CODE REVIEW

File: .github/workflows/build_profiler.yml [hunk 1]
```diff
@@ -109,6 +120,29 @@ jobs:
...
+        run: |
+          choco install OpenCppCoverage -y
+          echo "C:\Program Files\OpenCppCoverage" >> $env:GITHUB_PATH
+
+      - name: Generate Report
+        id: generate_test_report
+        shell: cmd
+        run: |
+            cd ${{ env.profiler_path }}
+            OpenCppCoverage.exe --sources Profiler --excluded_sources rapidxml --excluded_sources Profiler\SystemCalls.h --excluded_sources test --modules NewRelic\Profiler --export_type cobertura:TestResults\profiler.xml -- "vstest.console.exe" "CommonTest\bin\x86\Release\CommonTest.dll" "ConfigurationTest\bin\x86\Release\ConfigurationTest.dll" "LoggingTest\bin\x86\Release\LoggingTest.dll" "MethodRewriterTest\bin\x86\Release\MethodRewriterTest.dll" "SignatureParserTest\bin\x86\Release\SignatureParserTest.dll" "Sicily\SicilyTest\bin\x86\Release\SicilyTest.dll"
```
Comment (chynesNR) [inline .github/workflows/build_profiler.yml:139]: This is a little hard to parse, but it's running all the different unit test DLLs, asking for coverage for anything in our Profiler directory, and excludes rapidxml (which is third party code) and SystemCalls.h, since we don't want to call native libraries (creating directories, etc) for unit tests.

**Review by tippmar-nr — COMMENTED** [review]
Nice work!

**Review by nrcventura — APPROVED** [review]
